New in doc: May 2025 (#2765)
ibexa/documentation-developer: docs/resources/new_in_doc.md

This page contains recent highlights and notable changes in [[= product_name =]] documentation.

## May 2025

> ### v4.6.20

Reviewer: I would have keep "Infrastructure and maintenance" but this is interestingly highlighting the release, release even visible in the ToC.

    - [`Ibexa\Contracts\TwigComponents\Event\RenderSingleEvent`](https://doc.ibexa.co/en/latest/api/php_api/php_api_reference/classes/Ibexa-Contracts-TwigComponents-Event-RenderSingleEvent.html)
    - [`Ibexa\Contracts\TwigComponents\Exception\InvalidArgumentException`](https://doc.ibexa.co/en/latest/api/php_api/php_api_reference/classes/Ibexa-Contracts-TwigComponents-Exception-InvalidArgumentException.html)
    - [`Ibexa\Contracts\TwigComponents\Renderer\RendererInterface`](https://doc.ibexa.co/en/latest/api/php_api/php_api_reference/classes/Ibexa-Contracts-TwigComponents-Renderer-RendererInterface.html)

### Requirements update

- Updated [requirements](https://doc.ibexa.co/en/latest/getting_started/requirements/) for [[= product_name =]] 4.6: MySQL 8.4, Node 20 and Node 22 are now officially supported

### AI Actions Connect handler

- Documented [how to use the `ibexa:connect:init-custom-property-structures` command](https://doc.ibexa.co/en/latest/ai_actions/install_ai_actions/#initiate-integration) to integrate [AI actions](https://doc.ibexa.co/en/latest/ai_actions/ai_actions_guide/) with [Ibexa Connect](https://doc.ibexa.co/projects/connect/en/latest/)

> ### v4.6.20

Reviewer 2: Could you please put it as a first bullet? I think a new release is the most important

    - [`Ibexa\Contracts\TwigComponents\ComponentRegistryInterface`](https://doc.ibexa.co/en/latest/api/php_api/php_api_reference/classes/Ibexa-Contracts-TwigComponents-ComponentRegistryInterface.html)
    - [`Ibexa\Contracts\TwigComponents\Event\RenderGroupEvent`](https://doc.ibexa.co/en/latest/api/php_api/php_api_reference/classes/Ibexa-Contracts-TwigComponents-Event-RenderGroupEvent.html)
    - [`Ibexa\Contracts\TwigComponents\Event\RenderSingleEvent`](https://doc.ibexa.co/en/latest/api/php_api/php_api_reference/classes/Ibexa-Contracts-TwigComponents-Event-RenderSingleEvent.html)
    - [`Ibexa\Contracts\TwigComponents\Exception\InvalidArgumentException`](https://doc.ibexa.co/en/latest/api/php_api/php_api_reference/classes/Ibexa-Contracts-TwigComponents-Exception-InvalidArgumentException.html)
    - [`Ibexa\Contracts\TwigComponents\Renderer\RendererInterface`](https://doc.ibexa.co/en/latest/api/php_api/php_api_reference/classes/Ibexa-Contracts-TwigComponents-Renderer-RendererInterface.html)

### Getting started

> - Updated product version [requirements](https://doc.ibexa.co/en/latest/getting_started/requirements/) and [installation instructions](https://doc.ibexa.co/en/latest/getting_started/install_ibexa_dxp)

Reviewer 2: ```suggestion
```
This was done for v5, which is not public yet - we should remove this section, nothing changed for released versions

### Requirements update

> - Updated [requirements](https://doc.ibexa.co/en/latest/getting_started/requirements/) for [[= product_name =]]: MySQL 8.4, Node 20 and Node 22 are now officially supported

Reviewer 2: ```suggestion
- Updated [requirements](https://doc.ibexa.co/en/latest/getting_started/requirements/) for [[= product_name =]] 4.6: MySQL 8.4, Node 20 and Node 22 are now officially supported
```

> ### Ibexa AI handler

Reviewer 2: ```suggestion
### AI Actions Connect handler
```

> - New `Ibexa AI handler` mentioned in AI Actions documentation:
>     - [Create custom Action Handler](https://doc.ibexa.co/en/latest/ai_actions/extend_ai_actions/#create-custom-action-handler)
>     - [Initiate integration](https://doc.ibexa.co/en/latest/ai_actions/install_ai_actions/#initiate-integration)

Reviewer 2: ```suggestion
- Documented [how to use the `ibexa:connect:init-custom-property-structures` command](https://doc.ibexa.co/en/latest/ai_actions/install_ai_actions/#initiate-integration) to integrate [AI actions](https://doc.ibexa.co/en/latest/ai_actions/ai_actions_guide/) with [Ibexa Connect](https://doc.ibexa.co/projects/connect/en/latest/)
```

The `Ibexa AI handler` is the name of a custom property that I'd treat as an implementation detail, without exposing it here.

This change is like a bugfix to the doc - we missed this when creating the documentation and it's a mandatory step in AI + Connect integration

### Contributions

#### Order management

- Updated [Order management API](https://doc.ibexa.co/en/latest/commerce/order_management/order_management_api) documentation

#### Reverse proxy

- Fixed [Reverse proxy](https://doc.ibexa.co/en/latest/infrastructure_and_maintenance/cache/http_cache/reverse_proxy) documentation:

>     - [link to []Varnish VCL xkey example](https://github.com/ibexa/http-cache/blob/4.6/docs/varnish/vcl/varnish5.vcl) link

Reviewer: Remove brackets:
```suggestion
    - [link to Varnish VCL xkey example](https://github.com/ibexa/http-cache/blob/4.6/docs/varnish/vcl/varnish5.vcl) link
```

Reviewer: I don't know if it worth putting "xkey" into code style:
```suggestion
    - [link to Varnish VCL `xkey` example](https://github.com/ibexa/http-cache/blob/4.6/docs/varnish/vcl/varnish5.vcl) link
```

> ### Contributions
> 
> #### Order management
> 
> - Updated [Order management API](https://doc.ibexa.co/en/latest/commerce/order_management/order_management_api) documentation
> 
> #### Reverse proxy
> 
> - Fixed [Reverse proxy](https://doc.ibexa.co/en/latest/infrastructure_and_maintenance/cache/http_cache/reverse_proxy) documentation:
>     - [link to []Varnish VCL xkey example](https://github.com/ibexa/http-cache/blob/4.6/docs/varnish/vcl/varnish5.vcl) link
>     - [`ez_purge_acl`](https://github.com/ibexa/http-cache/blob/4.6/docs/varnish/vcl/varnish5.vcl#L174) path

Reviewer 2: ```suggestion
### Contributions

- We want to thank [todomagichere](https://github.com/todomagichere) and [hgiesenow](https://github.com/hgiesenow) for their help in improving the [Order management API](https://doc.ibexa.co/en/latest/commerce/order_management/order_management_api) and [Reverse proxy](https://doc.ibexa.co/en/latest/infrastructure_and_maintenance/cache/http_cache/reverse_proxy) documentation pages
```

The changes were mostly about fixing existing errors in the doc, so I wouldn't expand on what has been changed - but anyway I think it's worth expressing our gratutide for the contributions
